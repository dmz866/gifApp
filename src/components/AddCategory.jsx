import PropTypes from 'prop-types';
import { useState } from "react";

export const AddCategory = ({ addCategory }) => {
	const [inputValue, setInputValue] = useState('');
	const onInputChange = (e) => setInputValue(e.target.value);
	const onSubmit = (e) => {
		e.preventDefault();

		if (inputValue.trim()) {
			addCategory(inputValue);
		}

		setInputValue('');
	};

	return (
		<form onSubmit={onSubmit} aria-label='form'>
			<input type='text' placeholder='Buscar gifs' value={inputValue} onChange={onInputChange} />
		</form>
	)
};

AddCategory.propTypes = {
	addCategory: PropTypes.func.required
};
