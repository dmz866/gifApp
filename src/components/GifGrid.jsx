import { useEffect, useState } from "react";
import { getGifs } from '../helpers/gifHelper';

export const GifGrid = ({ category }) => {
	const [images, setImages] = useState([]);
	const getImages = async () => {
		if (!category) return;

		const images = await getGifs(category);
		setImages(images);
	};

	useEffect(() => {
		getImages();
	}, []);

	return (
		<div>
			<h3>{category}</h3>
			<ol>
				{images.map(({ id, titulo }) => (
					<li key={id}>{titulo}</li>
				))}
			</ol>
		</div>
	)
};
