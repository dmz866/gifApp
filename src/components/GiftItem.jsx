import PropTypes from 'prop-types';

export const GiftItem = ({ titulo, url }) => {
	return (
		<div className="card">
			<img src={url} alt={titulo} />
			<p>{titulo}</p>
		</div>
	)
};

GiftItem.protoTypes = {
	titulo: PropTypes.string.required,
	url: PropTypes.string.required
};