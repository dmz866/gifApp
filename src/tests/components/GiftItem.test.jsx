import { render } from "@testing-library/react";
import { GiftItem } from "../../components";

describe('Test GiftItem', () => {
	test('debe hacer match con el snapshot', () => {
		const { container } = render(<GiftItem titulo='tituloTest' url='urlTest' />);

		expect(container).toMatchSnapshot();
	});
});