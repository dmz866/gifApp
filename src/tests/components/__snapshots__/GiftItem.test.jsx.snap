// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GiftItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="tituloTest"
      src="urlTest"
    />
    <p>
      tituloTest
    </p>
  </div>
</div>
`;

exports[`Test GiftItem debe hacer match con el snapshot 2`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="tituloTest"
      src="urlTest"
    />
    <p>
      tituloTest
    </p>
  </div>
</div>
`;

exports[`Test GiftItem debe mostrar la imagen  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="tituloTest"
      src="urlTest"
    />
    <p>
      tituloTest
    </p>
  </div>
</div>
`;
